### PR TITLE
Well Log Viewer: 'gradientfill' plot

### PR DESCRIPTION
**Pull request by Vladimir-Kokin**
- new plot GradientFillPlot
- color-table JSON file
- color table utility functions
- new Add Plot Dialog
- fix Scroller: move it out of WellLogView to WellLogViewer
- fix the component to have variable size
- use existing  'differential' plot (as a first step for #596)
- use 'log' scale 

For gradient fill plot type see #595 
For color table see #561

### CODE REVIEW

File: react/src/lib/components/WellLogViewer/components/LocalMenus.tsx [hunk 1]
```diff
@@ -0,0 +1,328 @@
...
+        this.addTrack = this.addTrack.bind(this);
+        this.removeTrack = this.removeTrack.bind(this);
+    }
+    componentDidUpdate(prevProps: SimpleMenuProps): void {
+        if (this.props.anchorEl !== prevProps.anchorEl) {
+            this.setState((_state, props) => {
+                return { anchorEl: props.anchorEl };
+            });
+        }
+        /*if (
```
Comment (shadab-skhan) [inline react/src/lib/components/WellLogViewer/components/LocalMenus.tsx:62]: Redundant code

File: react/src/lib/components/WellLogViewer/components/WellLogView.tsx [hunk 2]
```diff
@@ -6,540 +6,51 @@ import {
...
 import { ScaleTrackOptions } from "@equinor/videx-wellog/dist/tracks/scale/interfaces";
-import { GraphTrackOptions } from "@equinor/videx-wellog/dist/tracks/graph/interfaces";
 
 import {
     OverlayClickEvent,
     OverlayMouseMoveEvent,
     OverlayMouseExitEvent,
     OverlayRescaleEvent,
-    LogControllerResizeEvent,
+    //LogControllerResizeEvent,
```
Comment (shadab-skhan) [inline react/src/lib/components/WellLogViewer/components/WellLogView.tsx:18]: Commented code

File: react/src/lib/components/WellLogViewer/components/WellLogView.tsx [hunk 3]
```diff
@@ -1015,128 +577,76 @@ class WellLogView extends Component<Props, State> implements WellLogController {
...
+    removeGraphTrackPlot(track: GraphTrack, name: string): void {
+        removeGraphTrackPlot(this, track, name);
+        if (this.logController) this.logController.updateTracks();
+        this.setInfo();
     }
-    setScrollTrack(): void {
-        const iFrom = this._newScrollPos(this.state.scrollTrackPos);
+
+    scrollTrack(): void {
+        /*
```
Comment (shadab-skhan) [inline react/src/lib/components/WellLogViewer/components/WellLogView.tsx:620]: Commented code

File: react/src/lib/components/WellLogViewer/utils/color-table.ts [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+/*
+  get HTML string in #xxxxxx format with color value given in the first 3 elements of array
+*/
+export function colorToString(
+    color: [number, number, number] | undefined, // [r,g,b]
+    cDefault: string
+): string {
+    if (!color) return cDefault;
+    // old code
+    // const p =
```
Comment (shadab-skhan) [inline react/src/lib/components/WellLogViewer/utils/color-table.ts:29]: can remove it, if not required in future.

File: react/src/lib/components/WellLogViewer/utils/tracks.ts [hunk 5]
```diff
@@ -5,22 +5,20 @@ import {
     GraphTrack,
 } from "@equinor/videx-wellog";
 
-import { AreaPlotOptions } from "@equinor/videx-wellog/dist/plots/interfaces";
-
-import { GraphTrackOptions } from "@equinor/videx-wellog/dist/tracks/graph/interfaces";
+import {
+    LegendInfo /*,DifferentialPlotLegendInfo*/,
+} from "@equinor/videx-wellog/dist/plots/legend/interfaces";
+// import { DifferentialPlotOptions/*,AreaPlotOptions*/ } from "@equinor/videx-wellog/dist/plots/interfaces";
```
Comment (shadab-skhan) [inline react/src/lib/components/WellLogViewer/utils/tracks.ts:11]: commented import

File: react/src/lib/components/WellLogViewer/utils/tracks.ts [hunk 6]
```diff
@@ -5,22 +5,20 @@ import {
...
-export interface ExtPlotOptions extends AreaPlotOptions {
-    legendInfo: () => {
-        label: string;
-        unit: string;
-    };
-}
-
 import { PlotConfig } from "@equinor/videx-wellog/dist/tracks/graph/interfaces";
-
-//import { GraphTrackOptions } from "@equinor/videx-wellog";
```
Comment (shadab-skhan) [inline react/src/lib/components/WellLogViewer/utils/tracks.ts:23]: commented import

File: react/src/lib/components/WellLogViewer/utils/tracks.ts [hunk 7]
```diff
@@ -293,58 +468,111 @@ export function addTrackPlot(
...
+    for (const plot of plots) {
+        if (plot === _plot) {
+            plots.splice(index, 1);
+            break;
+        }
+        index++;
+    }
+
+    /*
+    (track as GraphTrack).prepareData(); //
```
Comment (shadab-skhan) [inline react/src/lib/components/WellLogViewer/utils/tracks.ts:566]: commented code

**Review by shadab-skhan — APPROVED** [review]
Change is good.
There are few redundant/commented code blocks, we can clean up if not required.
Also, there are few comments marked to-do, do we need to create an issue/task for it?

File: react/src/lib/components/WellLogViewer/utils/color-table.ts (hunk 4 above)
Comment (Vladimir-Kokin) [reply react/src/lib/components/WellLogViewer/utils/color-table.ts:29]: Oh! No!  The code is not so obvious. So it should have вуефшдув вщсгьутефешщт

**Comment by Vladimir-Kokin** [comment]
I have removed some commented code.

**Pull request merged**